templates/srTakeover_hook.j2

```diff
@@ -0,0 +1,113 @@
+
+import os, time, dbapi
+
+from hdb_ha_dr.client import HADRBase, Helper
+
+class srTakeover(HADRBase):
+    def __init__(self, *args, **kwargs):
+        # delegate construction to base class
+        super(srTakeover, self).__init__(*args, **kwargs)
+
+    def about(self):
+        return {"provider_company" : "SUSE",
+                "provider_name" : "srTakeover", # provider name = class name
+                "provider_description" : "Replication takeover script to set parameters to default.",
+                "provider_version" : "1.0"}
+
+    def startup(self, hostname, storage_partition, system_replication_mode, **kwargs):
+        self.tracer.debug("enter startup hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        self.tracer.info("leave startup hook")
+        return 0
+
+    def shutdown(self, hostname, storage_partition, system_replication_mode, **kwargs):
+        self.tracer.debug("enter shutdown hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        self.tracer.info("leave shutdown hook")
+        return 0
+
+    def failover(self, hostname, storage_partition, system_replication_mode, **kwargs):
+        self.tracer.debug("enter failover hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        self.tracer.info("leave failover hook")
+        return 0
+
+    def stonith(self, failingHost, **kwargs):
+        self.tracer.debug("enter stonith hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        # e.g. stonith of params["failed_host"]
+        # e-g- set vIP active
+        self.tracer.info("leave stonith hook")
+        return 0
+
+    def preTakeover(self, isForce, **kwargs):
+        """Pre takeover hook."""
+        self.tracer.info("%s.preTakeover method called with isForce=%s" %
+        (self.__class__.__name__, isForce))
+        if not isForce:
+            # run pre takeover code
+            # run pre-check, return != 0 in case of error => will abort takeover
+            return 0
+        else:
+            # possible force-takeover only code
+            # usually nothing to do here
+            return 0
+
+    def postTakeover(self, rc, **kwargs):
+        """Post takeover hook."""
+        self.tracer.info("%s.postTakeover method called with rc=%s" % (self.__class__.__name__, rc))
+        if rc == 0:
+            # normal takeover succeeded
+            conn = dbapi.connect('localhost',dbport,dbuser,dbpwd)
+            cursor = conn.cursor()
+            cursor.execute(stmnt1)
+            cursor.execute(stmnt2)
+            return 0
+        elif rc == 1:
+            # waiting for force takeover
+            conn = dbapi.connect('localhost',dbport,dbuser,dbpwd)
+            cursor = conn.cursor()
+            cursor.execute(stmnt1)
+            cursor.execute(stmnt2)
+            return 0
+        elif rc == 2:
+            # error, something went wrong
+            return 0
+{% endif %}
+{% endfor %}
```

Comment:
I think we can finish the prim `for` loop just after initializing the params, right? To avoid extra job

Reply:
Hi @arbulu89 the variables are defined and used in the if condition under prim for loop. I will check if there is a better way to avoid extra work

Reply:
What I only mean is that we don't need to have the whole script inside the for loops. We need the loops to get the variables data, that's it. That's why we could close the loops just after getting the needed data (after line 20 of the file)